Framework/Core/include/Framework/AnalysisDataModel.h

```diff
@@ -641,20 +641,32 @@ using AmbiguousFwdTrack = AmbiguousFwdTracks::iterator;
 // HMPID information
 namespace hmpid
 {
-DECLARE_SOA_INDEX_COLUMN(Track, track);                  //! Track index
-DECLARE_SOA_COLUMN(HMPIDSignal, hmpidSignal, float);     //! Signal of the HMPID
-DECLARE_SOA_COLUMN(HMPIDDistance, hmpidDistance, float); //! Distance between the matched HMPID signal and the propagated track
-DECLARE_SOA_COLUMN(HMPIDNPhotons, hmpidNPhotons, short); //! Number of detected photons in HMPID
-DECLARE_SOA_COLUMN(HMPIDQMip, hmpidQMip, short);         //! Collected charge in the HMPID
+DECLARE_SOA_INDEX_COLUMN(Track, track);                            //! Track index
+DECLARE_SOA_COLUMN(HMPIDSignal, hmpidSignal, float);               //! Signal of the HMPID
+DECLARE_SOA_COLUMN(HMPIDXTrack, hmpidXTrack, float);               //! Extrapolated track point x coordinate
+DECLARE_SOA_COLUMN(HMPIDYTrack, hmpidYTrack, float);               //! Extrapolated track point y coordinate
+DECLARE_SOA_COLUMN(HMPIDXMip, hmpidXMip, float);                   //! Matched MIP track point x coordinate
+DECLARE_SOA_COLUMN(HMPIDYMip, hmpidYMip, float);                   //! Matched MIP track point y coordinate
+DECLARE_SOA_COLUMN(HMPIDNPhotons, hmpidNPhotons, int);             //! Number of detected photons in HMPID
+DECLARE_SOA_COLUMN(HMPIDQMip, hmpidQMip, float);                   //! Matched MIP cluster charge
+DECLARE_SOA_COLUMN(HMPIDClusSize, hmpidClusSize, int);             //! Matched MIP cluster size
+DECLARE_SOA_COLUMN(HMPIDMom, hmpidMom, float);                     //! Matched MIP cluster size
+DECLARE_SOA_COLUMN(HMPIDPhotsCharge, hmpidPhotsCharge, float[10]); //! Photon cluster charge
 } // namespace hmpid
 
 DECLARE_SOA_TABLE(HMPIDs, "AOD", "HMPID", //! HMPID information
                   o2::soa::Index<>,
                   hmpid::TrackId,
                   hmpid::HMPIDSignal,
-                  hmpid::HMPIDDistance,
+                  hmpid::HMPIDXTrack,
+                  hmpid::HMPIDYTrack,
+                  hmpid::HMPIDXMip,
+                  hmpid::HMPIDYMip,
                   hmpid::HMPIDNPhotons,
-                  hmpid::HMPIDQMip);
+                  hmpid::HMPIDQMip,
+                  hmpid::HMPIDClusSize,
+                  hmpid::HMPIDMom,
+                  hmpid::HMPIDPhotsCharge);
```

Comment:
This breaks the Run 2 converted data which is written in the current format. 
You have to create a version 1 of the table and provide a converter from version 0 to 1

Reply:
Hi @jgrosseo, thanks for the suggestion! Done!